src/Compilers/CSharp/Portable/FlowAnalysis/DefiniteAssignment.cs

```diff
@@ -1878,6 +1903,10 @@ public override BoundNode VisitParameter(BoundParameter node)
             {
                 CheckAssigned(node.ParameterSymbol, node.Syntax);
             }
+            else if (CurrentSymbol is SynthesizedRecordConstructor)
```

Comment:
is this redundant with the check in NoteRead?

Reply:
You're correct. 
The logic in `NoteRead` wouldn't do anything superfluous if we called it in an `else` here, but somehow it feels clearer to only call it for the case we need.